command/agent/job_endpoint.go

```diff
@@ -412,20 +402,10 @@ func (s *HTTPServer) jobUpdate(resp http.ResponseWriter, req *http.Request,
 		}
 	}
 
-	var region *string
-
-	// Region in http request query param takes precedence over region in job hcl config
-	if args.WriteRequest.Region != "" {
-		region = helper.StringToPtr(args.WriteRequest.Region)
-	}
-	// If 'global' region is specified or if no region is given,
-	// default to region of the node you're submitting to
-	if region == nil || args.Job.Region == nil ||
-		*args.Job.Region == "" || *args.Job.Region == api.GlobalRegion {
-		region = &s.agent.config.Region
-	}
+	requestRegion, jobRegion := regionForJob(
+		args.Job, args.WriteRequest.Region, s.agent.config.Region)
```

Comment:
the second argument to `regionForJob` is queryRegion, but `args.WriteRequest.Region` isn't the query parameter, it's the entry in the request body:
https://github.com/hashicorp/nomad/blob/master/api/jobs.go#L1057
https://github.com/hashicorp/nomad/blob/master/api/jobs.go#L1127

same comment for https://github.com/hashicorp/nomad/pull/8347/files#diff-873734b35e73e20c8685472523e5a532R155

it may be that `regionForJob` needs all of these, or that we perform an initial comparison of the request body and query params, according to their precedence.

Reply:
> but args.WriteRequest.Region isn't the query parameter, it's the entry in the request body:

_ohhhhh!_...

* So it looks like we set the client region from the job: [command/job_run.go#L181-L184](https://github.com/hashicorp/nomad/blob/master/command/job_run.go#L181-L184)
* Which _overrides_ the client region set by the `-region` flag: [command/job_run.go#L52](https://github.com/hashicorp/nomad/blob/master/command/job_run.go#L52):
* The client region gets set for in write request as a query param: [api/api.go#L636](https://github.com/hashicorp/nomad/blob/master/api/api.go#L636)
* And then the query param region gets parsed as `args.Region`: [command/agent/job_endpoint.go#L30](https://github.com/hashicorp/nomad/blob/master/command/agent/job_endpoint.go#L30)
* (And `args.WriteRequest.Region` is always going to be nil in the CLI case: [command/job_run.go#L249](https://github.com/hashicorp/nomad/blob/master/command/job_run.go#L249))

> it may be that `regionForJob` needs all of these, or that we perform an initial comparison of the request body and query params, according to their precedence.

Yeah, for testing purposes it'd probably be easier if we just pass it the `args.Region`. Will fix.